index.ts

```diff
@@ -21,29 +21,28 @@ export function* sinergia(
       const itemIterator: IterableIterator<any> = task(accumulator, item);
 
       yield new Promise(resolve => {
-        const step = (timestamp) => { // timestamp not used
+        const step = () => {
           const iteration = itemIterator.next();
 
-          if (!iteration.done) {
-            window.requestAnimationFrame(step);
-          }
-          else {
+          if (iteration.done) {
             if (actualOptions.debug) console.log(`item task is done with latest iteration`, iteration);
             accumulator = iteration.value;
-            window.cancelAnimationFrame(animToken);
             resolve();
           }
+          else {
+            animToken = requestAnimationFrame(step);
+          }
         };
 
-        animToken = window.requestAnimationFrame(step);
+        animToken = requestAnimationFrame(step);
       });
     }
 
     return { value: accumulator };
   } finally {
     // This block is called when sinergia is interrupted with `.return()`
 
-    if (animToken) window.cancelAnimationFrame(animToken);
```

Comment:
Any reason about `window` removal? Of course it's not necessary but I use on purpose because it makes explicit which functions are not universal